src/runtime_src/core/edge/drm/zocl/zocl_aie.c

```diff
@@ -430,6 +431,137 @@ zocl_aie_reset(struct drm_zocl_dev *zdev)
 	return 0;
 }
 
+int
+zocl_aie_graph_alloc_context(struct drm_zocl_dev *zdev, u32 gid, u32 ctx_code,
+	struct sched_client_ctx *client)
+{
+	struct list_head *cptr, *gptr;
+	struct zocl_graph_ctx_node *gnode;
+	struct sched_client_ctx *ctx;
+	unsigned long ctx_flags, graph_flags;
+	int ret;
+
+	spin_lock_irqsave(&zdev->exec->ctx_list_lock, ctx_flags);
+
+	list_for_each(cptr, &zdev->exec->ctx_list) {
+		ctx = list_entry(cptr, struct sched_client_ctx, link);
+
+		spin_lock_irqsave(&ctx->graph_list_lock, graph_flags);
+
+		list_for_each(gptr, &ctx->graph_list) {
+			gnode = list_entry(gptr, struct zocl_graph_ctx_node,
+			    link);
+
+			if (gnode->gid == gid) {
```

Comment:
Do you want to invert the check which will remove the deep indentation?
```
if (gnode-->gid != gid)
    continue;
```

Reply:
Done.